lib/matplotlib/backend_bases.py

```diff
@@ -212,10 +219,23 @@ class RendererBase(object):
 
     """
     def __init__(self):
+        self._id = None
         self._texmanager = None
 
         self._text2path = textpath.TextToPath()
 
+    @property
+    def _uid(self):
+        """
+        A lightweight id for unique-ification purposes.
+
+        Along with id(self), this combination should be unique enough to use as
+        part of a caching key.
+        """
+        if self._id is None:
+            self._id = next(_unique_renderer_id)
```

Comment:
I would just make this a regular attribute created at instantiation (and leave a comment in the ctor).  I'm not sure what making this a property buys you...